src/util/artifacts.js

```diff
@@ -14,12 +16,41 @@ export async function prepareArtifactsDir(
     const stats = await fs.stat(artifactsDir);
     if (!stats.isDirectory()) {
       throw new UsageError(
-        `--artifacts-dir=${artifactsDir} (this value is not a directory)`);
+        `--artifacts-dir="${artifactsDir}" exists but it is not a directory.`);
+    }
+    // If the artifactsDir already exists, check that we have the write permissions on it.
+    try {
+      await fs.access(artifactsDir, fs.W_OK);
+    } catch (accessErr) {
+      if (isErrorWithCode('EACCES', accessErr)) {
+        throw new UsageError(
+          `--artifacts-dir="${artifactsDir}" exists but the user lacks ` +
+          'permissions on it.');
+      } else {
+        throw accessErr;
+      }
     }
   } catch (error) {
-    if (isErrorWithCode('ENOENT', error)) {
-      log.debug(`Creating artifacts directory: ${artifactsDir}`);
-      await fs.mkdir(artifactsDir);
+    if (isErrorWithCode('EACCES', error)) {
+      // Handle errors when the artifactsDir cannot be accessed.
+      throw new UsageError(
+        `Cannot access --artifacts-dir="${artifactsDir}" because the user ` +
+        'lacks permissions.');
```

Comment:
I think this should include the original error to help the user pinpoint exactly what the access problem is. You can do it like:
````js
`Cannot access --artifacts-dir="${artifactsDir}" because the user ` +
`lacks permissions: ${error}`
````
Otherwise the permission error may not be completely obvious: the user may not own own the file, the read bit may not be set, the parent dir may not have an executable bit, etc.